pkg/hooks/loader.go

```diff
@@ -225,14 +244,34 @@ func (h *Hook) GetConfigMocks() ([]*models.Mock, error) {
 
 func (h *Hook) DeleteTcsMock(mock *models.Mock) bool {
 	isDeleted := h.tcsMocks.delete(mock.TestModeInfo)
+	h.matchedMocks = append(h.matchedMocks, mock.Name)
+	h.persistentMatchedMocks[mock] = true
 	return isDeleted
 }
 
 func (h *Hook) DeleteConfigMock(mock *models.Mock) bool {
 	isDeleted := h.configMocks.delete(mock.TestModeInfo)
+	h.matchedMocks = append(h.matchedMocks, mock.Name)
+	h.persistentMatchedMocks[mock] = true
 	return isDeleted
 }
 
+func (h *Hook) GetMatchedMocks() []string {
+	return h.matchedMocks
+}
+
+func (h *Hook) GetUsedMocks() ([]*models.Mock, error) {
+	var matchedMocks []*models.Mock
+	for matchedMock := range h.persistentMatchedMocks {
+		matchedMocks = append(matchedMocks, matchedMock)
+	}
+	return matchedMocks, nil
+}
+
```

Comment:
why maintain a map when an array can be maintained and array is used in the end?

Reply:
This map serves the purpose of set data structure, in case where addition of same entries happens an array would store copies but persistentMatchedMocks would store only one